toolchain/base/kind_switch.h

```diff
@@ -92,4 +167,11 @@ auto Cast(ValueT&& kind_switch_value) -> auto {
             false) {}                                                   \
   else [[maybe_unused]] CARBON_INTERNAL_KIND_LABEL(__LINE__)
 
+// Like `CARBON_KIND` but works with a type without a name, and does not make
+// the switch's value available in the case body as a result.
+#define CARBON_KIND_(type_without_name)            \
```

Comment:
Soliciting opinions on how to name this, or if we can/should make L164 work without a variable name somehow.

Reply:
Maybe don't add this, and use a variable name of `_`? That seems maybe a bit more in line with a pattern-like approach.

Reply:
Oh, I guess clang-tidy has an exception to allowing that name to be unused? Cool, done.

Reply:
Yeah, and there are special rules for it in C++26 (you can have multiple `_`s in the same scope).